back/boxtribute_server/business_logic/box_transfer/agreement/crud.py

```diff
@@ -179,6 +192,6 @@ def retrieve_transfer_agreement_bases(*, transfer_agreement, kind):
         TransferAgreementDetail, on=getattr(TransferAgreementDetail, f"{kind}_base")
     ).where(
         TransferAgreementDetail.transfer_agreement == transfer_agreement.id
-    ) or Base.select().where(
+    ).distinct() or Base.select().where(
```

Comment:
Why?

Reply:
If an agreement is between source bases 1&2 and target base 3, there's two rows in the TransferAgreementDetail table (assigned to the same agreement):
- one with source base 1 and target base 3 and
- one with source base 2 and target base 3
When I now query for all target bases of that agreement, it would return base 3 twice, hence `distinct` is needed.